src/common.c

```diff
@@ -445,40 +445,44 @@ get_mentions(gboolean whole_word, gboolean case_sensitive, const char* const mes
 gboolean
 call_external(gchar** argv, gchar** std_out, gchar** std_err)
 {
-    GError *spawn_error, *status_error;
-    gboolean spawn_result;
-    gint exit_status;
+    GError* spawn_error = NULL;
+    GError* exit_error = NULL;
+    gboolean is_successful;
+    gint wait_status;
 
     GSpawnFlags flags = G_SPAWN_SEARCH_PATH;
     if (std_out == NULL)
         flags |= G_SPAWN_STDOUT_TO_DEV_NULL;
     if (std_err == NULL)
         flags |= G_SPAWN_STDERR_TO_DEV_NULL;
 
-    spawn_result = g_spawn_sync(NULL, // Inherit the parent PWD.
-                                argv,
-                                NULL, // Inherit the parent environment.
-                                flags,
-                                NULL, NULL, // No func. before exec() in child.
-                                std_out, std_err,
-                                &exit_status, &spawn_error);
+    is_successful = g_spawn_sync(NULL, // Inherit the parent PWD.
+                                 argv,
+                                 NULL, // Inherit the parent environment.
+                                 flags,
+                                 NULL, NULL, // No func. before exec() in child.
+                                 std_out, std_err,
+                                 &wait_status, &spawn_error);
 
-    if (!spawn_result || !g_spawn_check_exit_status(exit_status, &status_error)) {
+    if (!is_successful) {
         gchar* cmd = g_strjoinv(" ", argv);
-        if (spawn_error && spawn_error->message) {
-            log_error("Spawning '%s' failed with '%s'.", cmd, spawn_error->message);
-            g_error_free(spawn_error);
-        } else if (status_error && status_error->message) {
-            log_error("Spawning '%s' failed with '%s'.", cmd, status_error->message);
-            g_error_free(status_error);
-            spawn_result = FALSE;
-        } else {
-            log_error("Spawning '%s' failed with.", cmd);
-        }
+        log_error("could not spawn '%s' with error '%s'", cmd, spawn_error->message);
```

Comment:
you're potentially de-referencing a `NULL`-pointer here
```suggestion
        log_error("could not spawn '%s' with error '%s'", cmd, spawn_error ? spawn_error->message : "Unknown, spawn_error is NULL");
```

Reply:
If that has to deal with that `g_spawn_sync` has weird behavior about setting errors then OK. Otherwise `is_successful == FALSE` leads to `spawn_error` to be set to some value hence `spawn_error` is not null.

Reply:
Yes, that's to deal with the weird behavior of `g_spawn_sync`.

Reply:
I believe we better submit a bug report for glib because the problem is that `spawn_error` stays uninitialized no matter what, or is set to a valid error otherwise. I haven't observed anything else except with a case when `spawn_error` is uninitialized in the first place.

Maybe we'll better add a custom error log message indicating there's something weird with glib instead of just doing `spawn_error ? spawn_error->message : "Unknown, spawn_error is NULL"`?

Reply:
IMO that'd be already a "custom error log message" since that will only pop up once this exact issue in glib is hit.

@jubalh you're the official maintainer, what do you think?

Reply:
I would prefer `"Unknown, spawn_error is NULL"` compared to a `its glibs fault!`. I think this error is clear enough, telling us whats going on and easy to grep for in code.

Reply:
This is missing

Reply:
I'll do this now